ibis_substrait/compiler/translate.py

```diff
@@ -860,6 +828,48 @@ def selection(
     return relation
 
 
+def _get_selections(op: ops.Selection) -> Sequence[ir.Column]:
+    if IBIS_4:
+        # projection / emit
+        selections = [
+            col
+            for sel in (x.to_expr() for x in op.selections)  # map ops to exprs
+            for col in (
+                sel.get_columns(sel.columns) if isinstance(sel, ir.TableExpr) else [sel]
+            )
+        ]
+    else:
+        # projection / emit
+        selections = [
+            col
+            for sel in op.selections
+            for col in (
+                sel.get_columns(sel.columns) if isinstance(sel, ir.TableExpr) else [sel]
+            )
+        ]
+
+    return selections
+
+
+def _find_parent_tables(op: ops.Selection) -> set[ir.Table]:
+    # TODO: settle on a better source table definition than "PhysicalTable with
+    # a schema"
+    if IBIS_4:
+        source_tables = {
+            t
+            for t in toposort(op).keys()
+            if isinstance(t, ops.PhysicalTable) and hasattr(t, "schema")
```

Comment:
What's a `PhysicalTable`?  I thought that meant "table on disk" / "source node" but I suspect I'm wrong?

Reply:
It's weirdly named -- it's the parent class of `UnboundTable`, `NamedTable`, `DatabaseTable` and others.  

Here it effectively fills in for "anything that probably has a `schema`" and then we double-check that.